src/grt/src/GlobalRouter.cpp

```diff
@@ -1880,6 +1882,56 @@ void GlobalRouter::connectPadPins(NetRouteMap& routes)
   }
 }
 
+void GlobalRouter::computePinOffsetAdjustments()
+{
+  for (auto net : block_->getNets()) {
+    if (pad_pins_connections_.find(net) != pad_pins_connections_.end()) {
+      for (GSegment& segment : pad_pins_connections_[net]) {
+        int tile_size = grid_->getTileSize();
+        int die_area_min_x = grid_->getXMin();
+        int die_area_min_y = grid_->getYMin();
+        int gcell_id_x
+            = floor((float) ((segment.init_x - die_area_min_x) / tile_size));
+        int gcell_id_y
+            = floor((float) ((segment.init_y - die_area_min_y) / tile_size));
+        if (segment.init_y == segment.final_y) {
+          for (int i = 0; i < gcells_offset_; i++) {
+            int curr_cap = fastroute_->getEdgeCapacity(gcell_id_x + i,
+                                                       gcell_id_y,
+                                                       gcell_id_x + i + 1,
+                                                       gcell_id_y,
+                                                       segment.init_layer);
+            curr_cap -= 1;
+            fastroute_->addAdjustment(gcell_id_x + i,
+                                      gcell_id_y,
+                                      gcell_id_x + i + 1,
+                                      gcell_id_y,
+                                      segment.init_layer,
+                                      curr_cap,
+                                      true);
+          }
+        } else if (segment.init_x == segment.final_x) {
+          for (int i = 0; i < gcells_offset_; i++) {
+            int curr_cap = fastroute_->getEdgeCapacity(gcell_id_x + i,
```

Comment:
Is this correct? If it follows the above code, the `+ i` should be on the `gcell_id_y`.

Reply:
yes it should be on the gcell_id_y, I will change that